src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerDriver.cs

```diff
@@ -1306,16 +1318,35 @@ private static void ComputeDeclarationsInNode(SemanticModel semanticModel, ISymb
                         executableCodeBlocks = declInNode.ExecutableCodeBlocks;
 
                         // Execute operation actions.
-                        if (shouldExecuteOperationActions && executableCodeBlocks.Any())
+                        if ((shouldExecuteOperationActions || shouldExecuteOperationBlockActions) && executableCodeBlocks.Any())
                         {
-                            var operationsToAnalyze = GetOperationsToAnalyze(executableCodeBlocks, semanticModel, cancellationToken);
-                            foreach (var analyzer in analysisScope.Analyzers)
+                            var operationBlocksToAnalyze = GetOperationBlocksToAnalyze(executableCodeBlocks, semanticModel, cancellationToken);
```

Comment:
Is `GetOperationBlocksToAnalyze` or `GetOperationsToAnalyze` an expensive computation? If so, you may want to consider caching the `operationBlocksToAnalyze` and `operationsToAnalyze` in `GetOrComputeDeclarationAnalysisData` so that for IDE execution, these get computed only once for all operation/operation block analyzers - current implementation will cause them to be re-computed for every analyzer in the IDE.